### PR TITLE
fix(no-ref): input-view

### PR DESCRIPTION
**Pull request by andriikamaldinov1**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/JsDaddy/ngx-copypaste/blob/develop/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/app/cards/cards.interface.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export interface ICard {
     id: number;
     anchor: string;
     isInput?: boolean;
+    isTextarea?: boolean;
```
[the remark this answers is not in the document]
Comment (andriikamaldinov1) [reply src/app/cards/cards.interface.ts:7]: if i use one varieable into this interface, my cards have input in all cards. Its not enough

[the remark this answers is not in the document]
Comment (xakeppok) [reply src/app/cards/cards.interface.ts:7]: please create enum for this

File: src/app/cards/cards.component.ts [hunk 2]
```diff
@@ -32,36 +34,29 @@ import { CopiedComponent } from '../shared/copied/copied.component';
...
     @Input() public cardDocs!: ICard[];
-    @Input() public cardExamples!: ICard[];
 
     @ViewChildren('cards') public cards!: QueryList<ElementRef>;
 
     public activeCardId = 1;
 
+    public readonly isShowInput = IsShowInput.INPUT;
+    public readonly isShowTextarea = IsShowInput.TEXTAREA;
+    public readonly hideInput = IsShowInput.NONE;
```
Comment (lachesse) [inline src/app/cards/cards.component.ts:50]: Change the naming to be more understandable

For example,
public readonly cardTypeInput = CardType.INPUT;

File: src/app/cards/cards.component.html [hunk 3]
```diff
@@ -24,7 +24,7 @@
             </div>
             <div class="dark-card bg-black">
                 <div class="dark-input column pd-15">
-                    <div class="input-wrapper h-input" *ngIf="card.isInput">
+                    <div class="input-wrapper h-input" *ngIf="card.isInput === cardTypeInput">
```
Comment (xakeppok) [inline src/app/cards/cards.component.html:27]: please use ngSwitch

File: src/app/cards/cards.component.html [hunk 4]
```diff
@@ -34,7 +34,9 @@
                             class="copy-position"
                             (click)="c1.copy()" />
                     </div>
-                    <div class="input-wrapper pd-15 h-input" *ngIf="!card.isInput">
+                    <div
+                        class="input-wrapper pd-15 h-input"
+                        *ngIf="card.isInput === cardTypeTextarea">
```
Comment (xakeppok) [inline src/app/cards/cards.component.html:39]: please USE enum. what is isInput ?

Comment (xakeppok) [inline src/app/cards/cards.component.html:39]: please use correct naming. isInput - boolean. but here it's enum.

File: src/app/cards/cards.enum.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const enum IsShowInput {
```
Comment (xakeppok) [inline src/app/cards/cards.enum.ts:1]: use correct naming!

File: src/app/cards/cards.component.ts [hunk 6]
```diff
@@ -32,36 +34,29 @@ import { CopiedComponent } from '../shared/copied/copied.component';
         ColorPipe,
         CopiedComponent,
     ],
-    providers: [ScrollService],
+    providers: [ScrollService, OnScrollService],
```
Comment (xakeppok) [inline src/app/cards/cards.component.ts:37]: for what you providing ScrollService if you did't use it?

File: src/app/cards/cards.component.ts [hunk 7]
```diff
@@ -32,36 +34,29 @@ import { CopiedComponent } from '../shared/copied/copied.component';
...
-                if (this.activeCardId !== Number(scrollIdCard)) {
-                    this.activeCardId = Number(scrollIdCard);
-                    this.router.navigate(['/'], {
-                        fragment: scrollIdCard,
-                    });
-                }
-            });
+        this.onScrollService.onScroll(this.cards);
+        this.router.events.pipe(takeUntil(this.unsubscribe$$)).subscribe(() => {
+            this.activeCardId = this.onScrollService.activeCardId;
```
Comment (xakeppok) [inline src/app/cards/cards.component.ts:59]: you have this in service. please RECHECK the whole component

File: src/app/cards/cards.component.ts [hunk 8]
```diff
@@ -36,32 +35,20 @@ import { CopiedComponent } from '../shared/copied/copied.component';
...
-    @Input() public cardExamples!: ICard[];
 
     @ViewChildren('cards') public cards!: QueryList<ElementRef>;
 
-    public activeCardId = 1;
-
+    public readonly activeCardId$: Observable<number> = inject(ScrollService).activeCard$;
+    public readonly cardTypeInput: CardType = CardType.INPUT;
+    public readonly cardTypeTextarea: CardType = CardType.TEXTAREA;
+    public readonly cardTypeHideInput: CardType = CardType.NONE;
```
Comment (xakeppok) [inline src/app/cards/cards.component.ts:46]: cardTypeNone. why it cardTypeHideInput ?

File: src/app/shared/copied/copied.component.ts [hunk 9]
```diff
@@ -9,4 +9,6 @@ import { AssetPipe } from '@libraries/asset/asset.pipe';
     templateUrl: './copied.component.html',
     styleUrls: ['./copied.component.scss'],
 })
-export class CopiedComponent {}
+export class CopiedComponent {
+    public readonly additionalPath = 'shared';
```
Comment (xakeppok) [inline src/app/shared/copied/copied.component.ts:13]: what is this? please look how it resolved in site

**Pull request merged**